src/material-experimental/mdc-slider/slider.ts

```diff
@@ -919,6 +929,31 @@ export class MatSlider
   _isRippleDisabled(): boolean {
     return this.disabled || this.disableRipple || !!this._globalRippleOptions?.disabled;
   }
+
+  /** Starts observing and updating the slider if the host changes its size. */
+  private _observeHostResize() {
+    if (typeof ResizeObserver === 'undefined' || !ResizeObserver) {
+      return;
+    }
+
+    // MDC only updates the slider when the window is resized which
+    // doesn't capture changes of the container itself. We use a resize
+    // observer to ensure that the layout is correct (see #24590).
+    this._ngZone.runOutsideAngular(() => {
+      // The callback will fire as soon as an element is observed and
+      // we only want to know after the initial layout.
```

Comment:
this is actually what `ResizeObserver` already does. The callback is not executed synchronously, but after initial layout is done, see https://medium.com/@paul_irish/requestanimationframe-scheduling-for-nerds-9c57f7438ef4

if you actually want to ignore the first layout (i.e. the first callback invocation), the comment is misleading

Reply:
I was playing around with resizing the browser and it definitely felt like it was emitting for each pixel.

Reply:
You could potentially link to https://www.w3.org/TR/resize-observer/#:~:text=Observation%20will%20fire-,when,-observation%20starts%20if